LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  TherapySettingsView.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 7/7/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import AVFoundation
+import HealthKit
+import LoopKit
+import SwiftUI
+
+public protocol TherapySettingsViewDelegate: class {
+    func gotoEdit(therapySetting: TherapySetting)
+    func save()
+    func cancel()
+}
+
+public class TherapySettingsViewModel: ObservableObject {
+    private var initialTherapySettings: TherapySettings
+    var therapySettings: TherapySettings
+
+    public init(therapySettings: TherapySettings = preview_therapySettings) {
+        self.therapySettings = therapySettings
+        self.initialTherapySettings = therapySettings
+    }
+    
+    /// Reset to original
+    func reset() {
+        therapySettings = initialTherapySettings
+    }
+}
+
+public struct TherapySettingsView: View, HorizontalSizeClassOverride {
+    @Environment(\.dismiss) var dismiss
+
+    weak var delegate: TherapySettingsViewDelegate?
+    
+    @ObservedObject var viewModel: TherapySettingsViewModel
+
+    @State var isEditing: Bool = false
+    
+    // Hack to test edit button below section or next to header
+    public enum DesignVersion {
+        case A, B, C
+        func cycle() -> DesignVersion {
+            switch self {
+            case .A: return .B
+            case .B: return .C
+            case .C: return .A
+            }
+        }
+    }
+    @State var editButtonDesignVersion: DesignVersion = .C
+
+    public enum PresentationMode {
+        case onboarding, settings
+    }
+    private let mode: PresentationMode
+    
+    public init(mode: PresentationMode = .settings, editButtonDesignVersion: DesignVersion = .C, viewModel: TherapySettingsViewModel) {
+        self.mode = mode
+        self.viewModel = viewModel
+        self.editButtonDesignVersion = editButtonDesignVersion
+    }
+    
+    public var body: some View {
+        switch mode {
+        case .settings: return AnyView(content())
+        case .onboarding: return AnyView(navigationContent())
+        }
+    }
+    
+    private func navigationContent() -> some View {
+        return NavigationView {
+            content()
+        }
+    }
+    
+    private func content() -> some View {
+        List {
+            correctionRangeSection
+            temporaryCorrectionRangesSection
+        }
+        .listStyle(GroupedListStyle())
+        .navigationBarTitle(Text(NSLocalizedString("Therapy Settings", comment: "Therapy Settings screen title")))
+        .navigationBarItems(leading: backOrCancelButton, trailing: editOrDoneButton)
+        .navigationBarBackButtonHidden(isEditing)
+        .environment(\.horizontalSizeClass, horizontalOverride)
+    }
+}
+
+typealias HKQuantityGuardrail = Guardrail<HKQuantity>
+
+extension TherapySettingsView {
+    
+    // TODO: Something better than this?
+    private var unit: HKUnit {
```

Comment:
I would be fine with adding a non-optional units field if that would help you on this screen

Reply:
Actually...that may not be the right way to go.  I will clean up the code to do something smarter.